airflow/contrib/executors/mesos_executor.py

```diff
@@ -118,10 +124,16 @@ def resourceOffers(self, driver, offers):
             driver.launchTasks(offer.id, tasks)
 
     def statusUpdate(self, driver, update):
-        logging.info("Task %s is in state %s, data %s", \
-            update.task_id.value, mesos_pb2.TaskState.Name(update.state), str(update.data))
-        
-        key = self.task_key_map[update.task_id.value]
+        logging.info("Task %s is in state %s, data %s",
+                     update.task_id.value, mesos_pb2.TaskState.Name(update.state), str(update.data))
+
+        try:
+            key = self.task_key_map[update.task_id.value]
+        except KeyError:
+            # The map may not contain an item if the framework re-registered after a failover.
+            # Discard these tasks.
+            logging.warn("Unrecognised task key %s" % update.task_id.value)
```

Comment:
This is not as big a problem as it looks because [get_event_buffer](https://github.com/airbnb/airflow/blob/master/airflow/executors/base_executor.py#L97) is only called by [BackfillJob](https://github.com/airbnb/airflow/blob/master/airflow/jobs.py#L641).